pallets/pools/src/lib.rs

```diff
@@ -166,104 +257,71 @@ where
 	}
 }
 
-/// Per-tranche and per-user order details.
-#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
-pub struct UserOrder<Balance, EpochId> {
-	pub invest: Balance,
-	pub redeem: Balance,
-	pub epoch: EpochId,
-}
-
-impl<Balance, EpochId> Default for UserOrder<Balance, EpochId>
-where
-	Balance: Zero,
-	EpochId: One,
-{
-	fn default() -> Self {
-		UserOrder {
-			invest: Zero::zero(),
-			redeem: Zero::zero(),
-			epoch: One::one(),
-		}
-	}
-}
-
-/// The result of epoch execution of a given tranch within a pool
-#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
-pub struct EpochDetails<BalanceRatio> {
-	pub invest_fulfillment: Perquintill,
-	pub redeem_fulfillment: Perquintill,
```

Comment:
I think we spoke about this before, but if we don't store this information anymore, how can we ensure (just using the `pools` pallet) that a scheduled update is only executed when all redemptions are fulfilled? Right now this is checking if `epochs[epochs.len - 1].redeem_fulfillment == 100%`.

Reply:
Just saw the code in the runtime lib. That works, but it does not guarantee the update invariant anymore solely using the `pools` pallet, as it now relies on the `investments` pallet. I think that's fine for now, but let's discuss later if this is fine or there might be an alternative approach.

Reply:
That is indeed true. I wanted to bring this in with the `UpdateGuard` changes. Seperate from registry and this integration. 

This should definitely be stored in the pools. Just wanted to give this a little more thought, how we actually store infos in the pools that ensure the pool itself can ensure all invariants itself.